src/Profiling/Profile.php

```diff
@@ -182,31 +214,30 @@ public function getFormattedData(Event $event): ?array
                     $function = $frame['class'] . '::' . $frame['function'];
                     $module = $frame['class'];
                 } elseif (isset($frame['function'])) {
-                    // {clousre}
+                    // {closure}
                     $function = $frame['function'];
                 } else {
                     // /index.php
                     $function = $file;
                 }
 
-                $frames[] = [
+                $stackFrames[] = $registerFrame([
```

Comment:
Not sure if this is applicable for PHP but in python, since we're able to use a triple composed of
- `abs_path`
- `lineno`
- `function`
as the map key. Normally `abs_path` and `lineno` suffice already but python allows you to patch over functions so we added `function` to dedupe that case.

Because of this, one thing we were able to do in python is to defer deriving the extra fields until later. Here, it looks like the work `$file` can be deferred until after checking if the frame doesn't exist.

Reply:
Wouldn't `abs_path` and `lineno` be not unique with recursive calls?

Reply:
We are showing from he callee perspective not the caller right? If so I think `abs_path` and `lineno` would indeed be enough since in PHP we don't have monkey patching... so that would remove the need for hashing and we can also indeed skip some more work if we already have the frame.

Reply:
> Wouldn't `abs_path` and `lineno` be not unique with recursive calls?

Why wouldn't it be unique? It's the same function, just called recursively. So we can use the same frame index to represent it.